lib/index.js

```diff
@@ -0,0 +1,240 @@
+  for(const id of mergeEventPeers) {
+    aggregate[id] = {id, weight: 1};
+  }
+
+  // TODO: weight previous electors more heavily to encourage continuity
+  // add consensusProof peers
+  for(const id of consensusProofPeers) {
+    if(id in aggregate) {
+      aggregate[id].weight = 3;
+    } else {
+      aggregate[id] = {id, weight: 2};
+    }
+  }
+  // add peers from the electorPool
+  for(const elector of electorPoolElectors) {
+    const electorType = [].concat(elector.type);
+    // electors defined in *any* way in the elector pool document should
+    // have the greatest weight
+    // FIXME: weight these like this?
+    // FIXME: is type === Continuity2017Peer a real thing?  If so, what do we
+    // do for that.
+    let weight = 100;
+    let recoveryElector = false;
+    if(electorType.includes(ELECTOR_TYPE_RECOVERY)) {
+      weight += 100;
+      recoveryElector = true;
+    }
+    if(electorType.includes(ELECTOR_TYPE_GUARANTOR)) {
+      weight += 100;
+    }
+    aggregate[elector.id] = {id: elector.id, recoveryElector, weight};
+  }
+
+  let electors = Object.values(aggregate);
+
+  let electorCount;
+  // the electorPool document may not be defined yet
+  if(electorPoolElectors.length > 0) {
+    // FIXME: `electorCount` in the ledgerConfiguration is used to specify a
+    // *maximum* number of electors correct?
+    // Therefore, electorCount is a maxiumum, so why would
+    // `electorPoolElectors.length` be considered a maximum?
+    electorCount = Math.min(
+      ledgerConfiguration.electorSelectionMethod.electorCount,
+      electorPoolElectors.length);
+  } else {
+    electorCount = ledgerConfiguration.electorSelectionMethod.electorCount;
+  }
+
+  // TODO: could optimize by only sorting tied electors if helpful
+  /*
+  // fill positions
+  let idx = -1;
+  for(let i = 0; i < electorCount; ++i) {
+    if(electors[i].weight > electors[i + 1].weight) {
+      idx = i;
+    }
+  }
+  // fill positions with non-tied electors
+  const positions = electors.slice(0, idx + 1);
+  if(positions.length < electorCount) {
+    // get tied electors
+    const tied = electors.filter(
+      e => e.weight === electors[idx + 1].weight);
+    // TODO: sort tied electors
+  }
+  }*/
+
+  const {blockHash} = latestBlockSummary.eventBlock.meta;
+
+  // break ties via sorting
+  const baseHashBuffer = Buffer.from(blockHash);
+  electors.sort((a, b) => {
+    // 1. sort descending by weight
+    if(a.weight !== b.weight) {
+      // FIXME: with current weights, this prevents elector cycling
+      //   if commented out, will force elector cycling, needs adjustment
+      return b.weight - a.weight;
+    }
+
+    // generate and cache hashes
+    // the hash of the previous block is combined with the elector id to
+    // prevent any elector from *always* being sorted to the top
+    a._hashBuffer = a._hashBuffer || xor(baseHashBuffer, Buffer.from(a.id));
+    b._hashBuffer = b._hashBuffer || xor(baseHashBuffer, Buffer.from(b.id));
```

Comment:
I suspect that buffer XOR is really slow. We did some other XOR thing elsewhere:

https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/blob/recovery-mode-electors/lib/consensus.js#L2156-L2169

Just a data point. I think we had some perf testing for this somewhere and it made a big difference --- but that was for events vs. electors, and there are far fewer electors (so it probably doesn't matter much).

Reply:
the XOR module does have an `inplace` implementation, but it did not seem to fit this application.  The computation being done here is not the same as the code you referenced.